resources/qml/delegates/PlayableMediaMessage.qml

```diff
@@ -0,0 +1,164 @@
+import QtQuick 2.6
+import QtQuick.Layouts 1.2
+import QtQuick.Controls 2.1
+import QtMultimedia 5.6
+
+import com.github.nheko 1.0
+
+Rectangle {
+	id: bg
+	radius: 10
+	color: colors.dark
+	height: content.height + 24
+	width: parent ? parent.width : undefined
+
+	Column { 
+		id: content
+		width: parent.width - 24
+		anchors.centerIn: parent
+
+		Rectangle {
+			id: videoContainer
+			visible: model.type == MtxEvent.VideoMessage
+			width: Math.min(parent.width, model.width ? model.width : 400) // some media has 0 as size...
+			height: width*model.proportionalHeight
+			Image {
+				anchors.fill: parent
+				source: model.thumbnailUrl.replace("mxc://", "image://MxcImage/")
+				asynchronous: true
+				fillMode: Image.PreserveAspectFit
+
+				VideoOutput {
+					anchors.fill: parent
+					fillMode: VideoOutput.PreserveAspectFit
+					source: media
+				}
+			}
+		}
+
+		RowLayout {
+			width: parent.width
+			Text {
+				id: positionText
+				text: "--:--:--"
+				color: colors.text
+			}
+			Slider {
+				Layout.fillWidth: true
+				id: progress
+				value: media.position
+				from: 0
+				to: media.duration
+
+				onMoved: media.seek(value)
+				//indeterminate: true
+				function updatePositionTexts() {
+					function formatTime(date) {
```

Comment:
Should probably make this respect the user's locale settings instead of using UTC time... or make it an option in the User Settings area.

Reply:
The issue is, that this actually shows a duration, not a time stamp. And I don't know how to format locale aware durations, so I just defaulted to the pretty universal format. I'll try looking into that some more at some point.